CONTRIBUTING.md

```diff
@@ -69,3 +69,87 @@ header.
 6. Create a pull request against the branch `master`.
 7. Be prepared to get some comments about your code and to modify it. Tip: Use
    `git rebase -i origin/master` to modify chains of commits.
+
+## Adding a test
+
+The test suite is a little ad-hoc, so the way tests work is different for each program being tested.
+
+Feel free to modify how the test scripts work, 
+
+### RGBASM
+
+Each `.asm` file corresponds to one test.
+RGBASM will be invoked on the `.asm` file.
+
+If a `.out` file exists, RGBASM's output (`println` etc.) must match its contents.
+If a `.err` file exists, RGBASM's errors (`warn`, errors, etc.) must match its contents.
+
+If a `.out.bin` file exists, the object file will be linked, and the generated ROM truncated to the length of the `.out.bin` file.
+After that, the ROM must match the `.out.bin` file.
+
+### RGBLINK
+
+Each `.asm` file corresponds to one test, or one *set* of tests.
+
+All tests begin by assembling the `.asm` file into an object file, which will be linked in various ways depending on the test.
+
+#### Simple tests
+
+These simply check that RGBLINK's output matches... *something*.
+A `.out` file **must** exist, and RGBLINK's output must match that file's contents.
+
+Additionally, if a `.out.bin` file exists, the `.gb` file generated by RGBLINK must match it.
+
+#### Linker script tests
+
+These allow applying various linker scripts to the same object file.
+If one or more `.link` files exist, whose names start the same as the `.asm` file, then each of those files correspond to one test.
+
+Each `.link` linker script **must** be accompanied by a `.out` file, and RGBLINK's output must match that file's contents when passed the corresponding linker script.
+
+#### Variant tests
+
+These allow testing RGBLINK's `-d`, `-t`, and `-w` flags.
+If one or more files called `<asm file name>-<flag>.out` or `<asm file name>-no-<flag>.out` exist, then each of them corresponds to one test.
```

Comment:
In general:
```
`foo-<var>-bar`
```
should be:
```
<code>foo-<i>&lt;var&gt;</i>-bar</code>
```
(RELEASE.md has some cases.)

Reply:
Fixed by 48479fee, though not in the two code *blocks* at the end for the reason explained in the commit message.